ports/raspberrypi/common-hal/wifi/Monitor.h

```diff
@@ -25,8 +25,11 @@
  * THE SOFTWARE.
  */
 
-#ifndef MICROPY_INCLUDED_ESPRESSIF_COMMON_HAL_WIFI_MONITOR_H
-#define MICROPY_INCLUDED_ESPRESSIF_COMMON_HAL_WIFI_MONITOR_H
+#pragma once
+
+#if !CIRCUITPY_WIFI
+#error wifi not in build
+#endif
```

Comment:
I think I would rather have this kind of checking in the .mk files. If you do provide these messages, I would make them say `CIRCUITPY_WIFI = 0, but this file is being included` for clarity. But the corresponding C files should not be being compiled. If they are not, then there would be link errors, which is how things would be caught.

Reply:
Correct, this is just an optimization for the build to fail at compile time instead of link time, because I'm impatient waiting for builds.
A typical scenario is I'm working on some module, and I sneak in a dependency on another module by including its header file. This compiles fine, but if the other module is not in the build config I happen to be using, I will get a link error. I think it would be hard to check this in .mk files since .mk files don't know what header files C files are including.

Reply:
Are you using `make -j12` etc.? I'd guess yes.

Reply:
I need a bigger computer. 😅

Reply:
Having `#error` all around the code is a +1 from me.
Linking can take 20s on x64 (-j12) and 4 minutes on my jacked (16% overclock) pi400.